index out of range panic on COUNT() query with WHERE clause

### Bug report

__System info:__ [Include InfluxDB version, operating system name, and other relevant details]

1.0.0-beta1
OSX

__Steps to reproduce:__

can't for the life of me get this to reproduce

__Actual behavior:__ [What actually happened]

Panic after running query:

```
select count(http_status_code) from nginx_access_log where http_status_code < 400
```

panic traceback:

```
[query] 2016/06/17 12:59:16 SELECT count(nginx_access_log) FROM telegraf.autogen.nginx_access_log WHERE http_status_code > 400
panic: runtime error: index out of range

goroutine 164 [running]:
panic(0x77ce40, 0xc82000a0a0)
	/usr/local/Cellar/go/1.6.2/libexec/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).createVarRefSeriesIterator(0xc82030a0c0, 0xc8209daa60, 0xc8200702a0, 0xc820799dc0, 0x2b, 0xc821195450, 0x3a00580, 0xc8207e7b90, 0xc820704b40, 0x1, ...)
	/Users/sparrc/ws/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:1148 +0x144c
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).createTagSetGroupIterators(0xc82030a0c0, 0xc8209daa60, 0xc8200702a0, 0xc820704a10, 0x1, 0x1, 0xc821195450, 0xc820704a30, 0x1, 0x1, ...)
	/Users/sparrc/ws/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:1068 +0x429
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).createTagSetIterators.func1(0xc8208a7c40, 0xc821740370, 0x2, 0x2, 0xc82030a0c0, 0xc8209daa60, 0xc8200702a0, 0xc821195450, 0xc82174ea80, 0x1)
	/Users/sparrc/ws/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:1023 +0x11a
created by github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).createTagSetIterators
	/Users/sparrc/ws/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:1024 +0x419
```

The same query without count() was working fine, as well as the count() query without the WHERE clause:

```
> select http_status_code from nginx_access_log where http_status_code > 400
...
> select count(http_status_code) from nginx_access_log
...
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -76,6 +76,7 @@ With this release the systemd configuration files for InfluxDB will use the syst
 - [#6834](https://github.com/influxdata/influxdb/pull/6834): Add port to all graphite log output to help with debugging multiple endpoints
 - [#6850](https://github.com/influxdata/influxdb/pull/6850): Modify the max nanosecond time to be one nanosecond less.
 - [#6824](https://github.com/influxdata/influxdb/issues/6824): Remove systemd output redirection.
+- [#6859](https://github.com/influxdata/influxdb/issues/6859): Set the condition cursor instead of aux iterator when creating a nil condition cursor.
 
 ## v0.13.0 [2016-05-12]
 

--- a/tsdb/engine/tsm1/engine.go
+++ b/tsdb/engine/tsm1/engine.go
@@ -1147,16 +1147,16 @@ func (e *Engine) createVarRefSeriesIterator(ref *influxql.VarRef, mm *tsdb.Measu
 				// If a field was requested, use a nil cursor of the requested type.
 				switch ref.Type {
 				case influxql.Float, influxql.AnyField:
-					aux[i] = &floatNilLiteralCursor{}
+					conds[i] = &floatNilLiteralCursor{}
 					continue
 				case influxql.Integer:
-					aux[i] = &integerNilLiteralCursor{}
+					conds[i] = &integerNilLiteralCursor{}
 					continue
 				case influxql.String:
-					aux[i] = &stringNilLiteralCursor{}
+					conds[i] = &stringNilLiteralCursor{}
 					continue
 				case influxql.Boolean:
-					aux[i] = &booleanNilLiteralCursor{}
+					conds[i] = &booleanNilLiteralCursor{}
 					continue
 				}
 			}

